CMake/resolve_dependency_modules/README.md

```diff
@@ -1,7 +1,47 @@
+# Dependency List
+Following is the list of libraries and their minimum version
+that Velox requires. Some of these libraries can be installed
+via a platform's package manager (eg. `brew` on MacOS).
+The versions of certain libraries is the default provided by 
+the platform's package manager. Some libraries can be bundled
+by Velox. See details on bundling below.
+
+| Library Name      | Minimum Version | Bundled? | 
+|-------------------|-----------------|----------|
+| ninja             | default         | No       |
+| ccache            | default         | No       |
+| icu4c             | default         | Yes      |
+| gflags            | default         | yes      |
+| glog              | default         | Yes      |
+| gtest (testing)   | default         | Yes      |
+| libevent          | default         | No       |
+| lz4               | default         | No       | 
+| snappy            | default         | No       |
+| lzo               | default         | No       |
+| xz                | default         | No       |
+| zstd              | default         | No       |
+| openssl           | default         | No       |
```

Comment:
We maybe want to set 3.0 as minimum due to 1.x being EOL, see post in slack.

Reply:
Let's change it when we make the actual changes.

Reply:
Let's mandate 3.0 for OpenSSL in a separate PR since that is always a can of worms.